src/main/scala/mesosphere/marathon/raml/PodStatusConversion.scala

```diff
@@ -92,23 +66,19 @@ trait PodStatusConversion {
 
     def containerStatus: Seq[ContainerStatus] = instance.tasks.map(t => Raml.toRaml((pod, t)))(collection.breakOut)
 
-    val derivedStatus: PodInstanceState = instance.state.status match {
-      case InstanceStatus.Created | InstanceStatus.Reserved => PodInstanceState.Pending
-      case InstanceStatus.Staging | InstanceStatus.Starting => PodInstanceState.Staging
-      case InstanceStatus.Error | InstanceStatus.Failed | InstanceStatus.Finished | InstanceStatus.Killed |
-        InstanceStatus.Gone | InstanceStatus.Dropped | InstanceStatus.Unknown | InstanceStatus.Killing |
-        InstanceStatus.Unreachable => PodInstanceState.Terminal
-      case InstanceStatus.Running =>
-        if (instance.state.healthy.getOrElse(true)) PodInstanceState.Stable else PodInstanceState.Degraded
-    }
+    val (derivedStatus: PodInstanceState, message: Option[String]) = podInstanceState(
+      instance.state.status, containerStatus)
 
     val networkStatus: Seq[NetworkStatus] = networkStatuses(instance.tasks.toVector)
 
     val resources: Option[Resources] = instance.state.status match {
-      case InstanceStatus.Staging | InstanceStatus.Starting | InstanceStatus.Running =>
+      case InstanceStatus.Staging | InstanceStatus.Starting | InstanceStatus.Running | InstanceStatus.Reserved |
```

Comment:
If you import InstanceStatus locally in this method, than this line becomes much more readable.
You should add `Killing` as well